src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,123 @@
+import { KeyboardEvent, ReactElement, ReactNode, useRef } from "react"
+import { TabList } from "./components/TabList"
+import { TabProps } from "./components/Tab"
+import { TabPanel } from "./components/TabPanel"
+import { Tab } from "./types"
+
+interface TabsProps {
+  children: ReactElement<TabProps> | Array<ReactElement<TabProps>>
+  currentTab: number
+  onTabChange: (index: number) => void
+}
+
+export function Tabs({ children, onTabChange, currentTab }: TabsProps) {
+  // We use a list of refs for tablist nodes so we can focus them programmatically
+  const tabListRefs = useRef<(HTMLLIElement | null)[]>([])
+
+  // Sometimes children is an array; sometimes it's a single node—let's make it an array
+  const childrenArray = Array.isArray(children) ? children : [children]
+
+  // Let's make sure the
+  let currentIndex =
+    currentTab < 0 || currentTab > childrenArray.length - 1 ? 0 : currentTab
```

Comment:
This is _potentially_ dangerous in a way that it could hide a programmer error.

If I try to render this:
```
<Tabs currentTab={999}>
  <Tab label="my tab">...</Tab>
</Tabs>
```

It's clear I'm doing something wrong, would I want for this to break / warn / log and/or do it's best to work?

Reply:
Yeah, I wasn't sure how to best handle this. It's definitely simpler to just tell the user to fix the problem. It would definitely simplify the implementation considerably.

Reply:
I've switched to just throwing errors when the `id` doesn't exist or it belongs to a disabled tab.